### PR TITLE
React refactor phase 1

### PR DESCRIPTION
**Pull request by chrisnojima**
Related to https://github.com/keybase/client/pull/1786

Phase 1: small cleanup
- [x] Deprecate base-react
- [x] Update package.json to use NODE_PATH desktop/node_modules
- [x] Update backtick concatting
- [x] Update protocol gen to export consts and not just export default (wrong es6 modules exports fixed in babel6)

@keybase/react-hackers

### CODE REVIEW

File: desktop/app/console-helper.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ export default function pipeLogs () {
       originalConsole[k].apply(console, args)
       if (args.length) {
         const out = output[k] || stdOutWriter
-        out(k + `: ${Date()} (${Date.now()}): ` + util.format.apply(util, args))
+        out(`${k}: ${Date()} (${Date.now()}): ${util.format.apply(util, args)}`)
```
Comment (chrisnojima) [inline desktop/app/console-helper.js]: let's just really use backticks

File: desktop/package.json [hunk 2]
```diff
@@ -7,10 +7,10 @@
     "start": "npm run build-dev && npm run start-cold",
     "start-hot": "HOT=true NODE_ENV=development node client.js",
     "start-cold": "NODE_ENV=development electron ./dist/main.bundle.js",
-    "build-dev": "NODE_ENV=production NO_SERVER=true DEBUG='express:*' node server.js",
-    "build-prod": "NODE_ENV=production webpack --config webpack.config.production.js --progress --profile --colors",
-    "package": "NODE_ENV=production node package.js",
-    "hot-server": "HOT=true NODE_ENV=production DEBUG='express:*' node server.js",
+    "build-dev": "NODE_PATH=`pwd`/node_modules NODE_ENV=production NO_SERVER=true DEBUG='express:*' node server.js",
```
Comment (chrisnojima) [inline desktop/package.json]: needed for module resolution for the various packagers

File: desktop/renderer/index.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
  * The main renderer. Holds the global store. When it changes we send it to the main thread which then sends it out to subscribers
  */
 
-import React, {Component} from '../../react-native/react/base-react'
```
Comment (chrisnojima) [inline desktop/renderer/index.js]: we can use react and not our base-react wrapper

File: protocol/bin/js.js [hunk 4]
```diff
@@ -36,9 +36,15 @@ function addEnums (prot, json) {
 }
 
 function write () {
-  var s = fs.createWriteStream('js/keybase_v1.js')
-  s.write('export default ' + JSON.stringify(protocols, null, 2).replace(/\"/g, '\'') + '\n')
-  s.close()
+  var stream = fs.createWriteStream('js/keybase_v1.js')
```
Comment (chrisnojima) [inline protocol/bin/js.js]: its an error to export a module like 
```
export default const { 
a: 1,
b: 2}
```

then do

```
import {a} from './blah'
```

This worked in babel 5 but won't in babel 6

File: react-native/react/menubar/index.render.desktop.js [hunk 5]
```diff
@@ -165,7 +165,7 @@ const ShowAll = props => {
   return <Row
     onClick={props.onClick}
     text='Show All'
-    iconStyle={{...SVGFolderIcon('file:///' + resolveAssets(`../react-native/react/images/see-more.svg`)), marginTop: 2}}
+    iconStyle={{...SVGFolderIcon(`file:///${resolveAssets('../react-native/react/images/see-more.svg')}`), marginTop: 2}}
```
Comment (chrisnojima) [inline react-native/react/menubar/index.render.desktop.js]: backticks support inner ticks

File: react-native/react/styles/common.js [hunk 6]
```diff
@@ -62,7 +60,7 @@ const buttons = {
   }
 }
 
-export default StyleSheet.create({
+export default sheet({
```
Comment (chrisnojima) [inline react-native/react/styles/common.js]: This whole thing is going away soon and we'll just use style-guide, but for now remove StyleSheet from the desktop side as it never really existed

**Comment by MarcoPolo** [comment]
:+1:

**Comment by chrisnojima** [comment]
I'm going to push this monday morning

**Pull request merged**